gems/aws-sdk-lambda/sig/client.rbs

```diff
@@ -0,0 +1,298 @@
+# WARNING ABOUT GENERATED CODE
+#
+# This file is generated. See the contributing guide for more information:
+# https://github.com/aws/aws-sdk-ruby/blob/version-3/CONTRIBUTING.md
+#
+# WARNING ABOUT GENERATED CODE
+
+module Aws
+  module Lambda
+    class Client < ::Seahorse::Client::Base
+      include ::Aws::ClientStubs
+
+      # https://docs.aws.amazon.com/sdk-for-ruby/v3/api/Aws/Lambda/Client.html#initialize-instance_method
+      def self.new: (?credentials: untyped, ?region: String, ?access_key_id: String, ?active_endpoint_cache: bool, ?adaptive_retry_wait_to_fill: bool, ?client_side_monitoring: bool, ?client_side_monitoring_client_id: String, ?client_side_monitoring_host: String, ?client_side_monitoring_port: Integer, ?client_side_monitoring_publisher: untyped, ?convert_params: bool, ?correct_clock_skew: bool, ?defaults_mode: String, ?disable_host_prefix_injection: bool, ?disable_request_compression: bool, ?endpoint: untyped, ?endpoint_cache_max_entries: Integer, ?endpoint_cache_max_threads: Integer, ?endpoint_cache_poll_interval: Integer, ?endpoint_discovery: bool, ?endpoint_provider: untyped, ?event_stream_handler: Proc, ?http_continue_timeout: Integer, ?http_idle_timeout: Integer, ?http_open_timeout: Integer, ?http_proxy: String, ?http_read_timeout: Integer, ?http_wire_trace: bool, ?ignore_configured_endpoint_urls: bool, ?input_event_stream_handler: Proc, ?log_formatter: untyped, ?log_level: Symbol, ?logger: untyped, ?max_attempts: Integer, ?on_chunk_received: Proc, ?on_chunk_sent: Proc, ?output_event_stream_handler: Proc, ?profile: String, ?raise_response_errors: bool, ?request_min_compression_size_bytes: Integer, ?retry_backoff: Proc, ?retry_base_delay: Float, ?retry_jitter: (:none | :equal | :full | ^(Integer) -> Integer), ?retry_limit: Integer, ?retry_max_delay: Integer, ?retry_mode: ("legacy" | "standard" | "adaptive"), ?sdk_ua_app_id: String, ?secret_access_key: String, ?session_token: String, ?ssl_ca_bundle: String, ?ssl_ca_directory: String, ?ssl_ca_store: String, ?ssl_timeout: Float, ?ssl_verify_peer: bool, ?stub_responses: untyped, ?token_provider: untyped, ?use_dualstack_endpoint: bool, ?use_fips_endpoint: bool, ?validate_params: bool) -> instance
+                  | (?Hash[Symbol, untyped]) -> instance
+
+
+      # https://docs.aws.amazon.com/sdk-for-ruby/v3/api/Aws/Lambda/Client.html#add_layer_version_permission-instance_method
+      def add_layer_version_permission: (action: Types::tLayerPermissionAllowedAction, layer_name: Types::tLayerName, principal: Types::tLayerPermissionAllowedPrincipal, statement_id: Types::tStatementId, version_number: Types::tLayerVersionNumber, ?organization_id: Types::tOrganizationId, ?revision_id: Types::tString) -> Types::AddLayerVersionPermissionResponse
```

Comment:
Is this the most useful part for operation signatures? Typically our customers use options with simple types and don't create Types structures. How does RBS resolve this? One improvement we could potentially make is translate types to simples, like "Types::tLayerName" to String. We may be able to do this nested through all API members. What do you think?

Reply:
I understand - we can only do the best we can with the current limitation. I defer to you (as a customer) for what you find most useful when developing with the SDK